File: core/src/main/java/com/ustadmobile/core/view/CustomFieldDetailView.java
package com.ustadmobile.core.view;


import com.ustadmobile.lib.db.entities.CustomField;

/**
 * Core View. Screen is for CustomDetailDetail's View
 */
public interface CustomFieldDetailView extends UstadView {


    // This defines the view name that is an argument value in the go() in impl.
    String VIEW_NAME = "CustomDetailDetail";

    //Any argument keys:
    String ARG_CUSTOM_FIELD_UID = "CustomFieldUid";

    void setDropdownPresetsOnView(String[] dropdownPresets);

    void setCustomFieldOnView(CustomField customField);

    void showOptions(boolean show);

    /**
     * Method to finish the screen / view.
     */
    void finish();


}

File: core/src/commonMain/kotlin/com/ustadmobile/staging/core/view/CustomFieldDetailView.kt
package com.ustadmobile.core.view


import androidx.paging.DataSource
import com.ustadmobile.core.db.UmProvider
import com.ustadmobile.lib.db.entities.CustomField
import com.ustadmobile.lib.db.entities.CustomFieldValueOption

/**
 * Core View. Screen is for CustomDetailDetail's View
 */
interface CustomFieldDetailView : UstadView {

    fun setDropdownPresetsOnView(dropdownPresets: Array<String>)
    fun setEntityTypePresetsOnView(entityTypePresets: Array<String>)

    fun setCustomFieldOnView(customField: CustomField)

    fun showOptions(show: Boolean)

    fun setListProvider(listProvider: DataSource.Factory<Int, CustomFieldValueOption>)

    /**
     * Method to finish the screen / view.
     */
    fun finish()

    companion object {


        // This defines the view name that is an argument value in the go() in impl.
        val VIEW_NAME = "CustomDetailDetail"

        //Any argument keys:
        val ARG_CUSTOM_FIELD_UID = "CustomFieldUid"
    }


}

